File: frontend/src/app/components/job-history/job-history.ts
import {Employee, EmployeeInterface} from "../employees/employee";
import {Job, JobInterface} from "../jobs/job";
import {Department, DepartmentInterface} from "../departments/department";

export interface JobHistoryInterface {
  employeeID: EmployeeInterface;
  startDate: string;
  endDate: string;
  jobID: JobInterface;
  departmentID: DepartmentInterface;
}

export class JobHistory {
  constructor(fields: Partial<JobHistoryInterface>) {
    this.employeeID = new Employee(fields.employeeID)
    this.startDate = fields.startDate;
    this.endDate = fields.endDate;
    this.jobID = new Job(fields.jobID);
    this.departmentID = new Department(fields.departmentID);
  }
  employeeID: Employee;
  startDate: string;
  endDate: string;
  jobID: Job;
  departmentID: Department;

}

File: src/main/java/pl/weronika/kurczyna/model/entity/JobHistory.java
package pl.weronika.kurczyna.model.entity;

import org.springframework.lang.Nullable;

import javax.persistence.*;
import java.sql.Date;

@Entity(name = "jobHistory")
@Table(name = "JOB_HISTORY")
@IdClass(CompositeKey.class)
public class JobHistory {
    @Id
    @JoinColumn(unique = true, name = "EMPLOYEE_ID")
    @ManyToOne(fetch = FetchType.EAGER)
    private Employee employeeID;
    @Id
    @Column(name = "START_DATE")
    private Date startDate;
    @Column(name = "END_DATE")
    private Date endDate;
    @JoinColumn(name = "JOB_ID")
    @ManyToOne(fetch = FetchType.EAGER)
    private Job jobID;
    @JoinColumn(name = "DEPARTMENT_ID")
    @Nullable
    @ManyToOne(fetch = FetchType.EAGER)
    private Department departmentID;

    public Employee getEmployeeID() {
        return employeeID;
    }

    public void setEmployeeID(Employee employeeID) {
        this.employeeID = employeeID;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public Job getJobID() {
        return jobID;
    }

    public void setJobID(Job jobID) {
        this.jobID = jobID;
    }

    @Nullable
    public Department getDepartmentID() {
        return departmentID;
    }

    public void setDepartmentID(@Nullable Department departmentID) {
        this.departmentID = departmentID;
    }
}
